#In a science research lab, combining two
#nuclear chemicals produces a maximum
#energy that is the product of the energy of the
#two chemicals  

#FIND TWO MAXIMUM NUMBERS FROM GIVEN LIST

In [22]:

n=int(input("enter n:"))
try:
    list=[]
    for i in range(n):
        list.append(int(input()))
except:
    print(list)
print(list)
list.sort()
print(list)    
print(list[-1]+list[-2])

enter n:4
12
34
54
123
[12, 34, 54, 123]
[12, 34, 54, 123]
177


<!-- bitwise opertaions
~ tilde :works on 8 bits   
& bitwise &
| bitwise or |
 left shift <<
 right shift >> -->

#bitwise operators  ~,&,|,^

In [24]:
 print(~10+~4)

-16


In [25]:
print(10 & 2)

2


In [26]:
print(10 | 2)

10


In [27]:
print(10 ^ 2)

8


In [28]:
print(10 << 2)

40


In [29]:
print(10 >> 2)

2


#floor , ceil , pow in math module

In [30]:
import math
print(math.floor(2.9))

2


In [31]:
import math
print(math.ceil(2.1))

3


3 ways to import the modules are
1)import math

2)import math as m

3)from math import sqrt,pow

In [32]:
import math as m
print(m.pow(2,3))

8.0


In [33]:
from math import pow ,sqrt
print(pow(3,3))
print(sqrt(4))

27.0
2.0


#program to find Even or odd

In [34]:
n=int(input("Enter a number:"))
print("Even") if(n%2==0) else print("Odd") 

Enter a number:5
Odd


#program to find sum of natural numbers

In [35]:
n = eval(input("enter the numbers:"))
print((n*(n+1))/2)

enter the numbers:4
10.0


 #program to find greatest three numbers

In [36]:
n1,n2,n3=10,20,30
max=n1 if n1>n2 else n2
max=n3 if n3>max else max
print(max)

30


#program to find greatest of two numbers

In [37]:
n1,n2=10,40
print(n1) if (n1>n2) else print(n2)

40


#program to find leap year or not

#two conditions to  be in leap year

# 1)it must be divisible by 400 

# 2)it must divisible by 4 and not by 100


In [38]:
year=int(input("Enter the year to check if its a leap year or not "))
print("leap year") if (year%400==0 or year%4==0 and year%100!=0) else print("not leap year")

Enter the year to check if its a leap year or not 2012
leap year


#program to find prime number 

In [39]:
n=27
flag=0

if n<2:
    flag=1
    
else:
    for i in range(2,int(pow(n,0.5)+1)):
        if n%i==0:
            flag=1
            break
print("Not Prime") if flag == 1 else print("Prime")            

            

Not Prime


#program to find prime number by recursion method

In [40]:
n=151
def checkPrime(n,i=2):
    if n == i:
        return True
    if n%i==0:
        return False
    if n<2:
        return False
    return checkPrime(n,i+1)

if checkPrime(n)==True:
    print("Prime")
else:
    print("Not Prime")

Prime


In [ ]:
for i in range(4):
    n=int(input("Enter  a number to check :"))
    
    flag=0
    if n<2:
        flag=1
    else:
        for i in range(2,int(pow(n,0.5)+1)):
            if n%i==0:
                flag=1
                break
    print("Not prime") if flag==1 else print("Prime")

Enter  a number to check :3
Prime


#program to print prime numbers in given range

In [1]:
Lower_Range  =  int(input("Enter the lower range:"))
Higher_Range =  int(input("Enter the Higher range:"))
for n in range(Lower_Range,Higher_Range+1):
    if n>1:
        for i in range(2,n):
            if n%i==0:
                break
        else:
            print(n,end=" ")

Enter the lower range:3
Enter the Higher range:23
3 5 7 11 13 17 19 23 

#program to find sum of given integer

In [4]:
n=int(input())
sum=0
while n>0:
    
    rem=n%10
    sum=sum+rem
    n=n//10
n=sum
def checkPrime(n,i=2):
    if n == i:
        return True
    if n%i==0:
        return False
    if n<2:
        return False
    return checkPrime(n,i+1)

if checkPrime(n)==True:
    print("GOOGLY")
else:
    print("NOT GOOGLY")

21
GOOGLY


#program to find revere number

In [44]:
n = int(input("Enter a number:"))
rev=0
while n>0:
    rev = rev * 10 + n % 10
    n=n//10
print(rev)
    

Enter a number:234
432


In [45]:
n=11222222222222222222222222222222222222211

def checkPalindrome(n):
    rev=0
    while n>0:
        rev=rev*10+n%10
        n=n//10
    return rev

print("Is Palindrome Number") if (checkPalindrome(n)==n) else print("Not a Palindrome")

Is Palindrome Number


#program to check given number is ARMSTRONG or not

In [46]:
n = 153
temp = n
order = len(str(n))
sum=0
for i in range(order):
    sum=sum+int(pow(temp%10,order))
    temp//=10
print("ARMSTRONG") if sum==n else print("NOT ARMSTRONG")
    

ARMSTRONG


#program to print Armstong numbers in given range

In [47]:
l=1
h=154
for n in range(l,h+1):
    order=len(str(n))
    sum=0
    temp=n
    while temp>0:
        sum=sum+int(pow(temp%10,order))
        temp//=10
    if n==sum:
        print(n,end=" ")

1 2 3 4 5 6 7 8 9 153 

#program to print fibonnaci series upto give number

In [10]:
n=int(input("Enter the number:"))
n1,n2=0,1
print("FIBONNACI SERIES:",n1,n2, end=" ")
for i in range(2,n):
    n3=n1+n2
    n1=n2
    n2=n3
    print(n3,end=" ")

Enter the number:10
FIBONNACI SERIES: 0 1 1 2 3 5 8 13 21 34 

#program to find factorial of given number

In [11]:
n=int(input("Enter the number:"))
fact=1
for i in range(1,n+1):
    fact=fact*i
print(fact,end=" ")

Enter the number:5
120 

#program to take input from user into list

In [2]:
list=[]
n=int(input("Enter the size of list :"))

for i in range(n):
    list.append(int(input()))
print(list)

Enter the size of list :4
1
2
3
4
[1, 2, 3, 4]


### program to check perfect number

In [10]:
for i in range(2):
    n=int(input("Enter the number to check perfect no. or not :"))
    sum=0
    for i in range(1,n):
        if n%i==0:
            sum=sum+i
    print("Perfect number") if sum==n else print("Not perfect number")

Enter the number to check perfect no. or not :4
Not perfect number
Enter the number to check perfect no. or not :6
Perfect number


#Program to check given number is AUTOMORPHIC or not, Ex Automorphic: 25 square=625 , so here last two digits of 625 is equal to #25  

In [2]:
n = int(input("Enter a number to check for Automorphic or no :"))
square = pow(n,2)
mod = pow(10,len(str(n)))
print("Automorphic number") if square%mod==n else print("Not Automorphic number")
    

Enter a number to check for Automorphic or no :12
Not Automorphic number


In [4]:
s = int(input("Enter a number to check for Automorphic or no :"))
square = pow(s,2)
mod = pow(10,len(str(s)))
print("Automorphic number") if square%mod==s else print("Not Automorphic number")
    

Enter a number to check for Automorphic or no :25
Automorphic number


#program to find HCF of two numbers

In [4]:
n1=10
n2=12
hcf=1
for i in range(1,min(n1,n2)):
               if n1%i==0 and n2%i==0:
                      hcf=i
                
print("hcf of",n1,"and",n2,"is :",hcf)
                     

hcf of 10 and 12 is : 2


#Program to check Harshad number   #harshad no.: No. which is divisible by sum of its no.
#Example : 21    , 2+1=3   , 21%3=0

In [12]:
number = int(input("Enter the Number to check :"))
n=number
sum=0
while number>0:
    rem = number%10
    sum+=rem
    number//=10
print(sum)
print("Its Harshad number") if n%sum==0 else print("Not Harshad number")

Enter the Number to check :21
3
Its Harshad number


In [11]:
number1 = int(input("Enter the Number to check :"))
n=number1
sum=0
while number1>0:
    rem = number1%10
    sum+=rem
    number1//=10
print(sum)
print("Its Harshad number") if n%sum==0 else print("Not Harshad number")

Enter the Number to check :23
5
Not Harshad number


#program to find LCM of two numbers

In [16]:
n1=24
n2=12
for i in range(max(n1,n2),1+(n1*n2)):
    if i%n1==i%n2==0:
         lcm=i
         break
print("lcm of ",n1 ,"and",n2,"is : ",lcm)

lcm of  24 and 12 is :  24


#Program to convert Binary to Decimal 

In [3]:
num=int(input("Enter the Binary number :"))
binaryValue = num
decimalValue = 0
base = 1
while num>0:
    rem = num%10
    decimalValue = decimalValue + rem*base
    num//=10
    base*=2
print("\nBinary value is {}\nDecimal value is {}".format(binaryValue,decimalValue))

Enter the Binary number :100

Binary value is 100
Decimal value is 4


#program to convert octal to decimal    no 8,9 octals

In [12]:
octalNum = 30
decimalNum = int(str(octalNum),8)
print("octal number is {}\nDecimal number is {}".format(octalNum,decimalNum))

octal number is 30
Decimal number is 24


#Program to replace all 0's whith 1's

In [3]:
n = int(input("Enter the numbers :"))
val = str(n)
replaced = val.replace('0','1')
print("Given number is {}\nConverted number is {}".format(n,replaced))

Enter the numbers :1202
Given number is 1202
Converted number is 1212


#Program to calculate the number of digits in a given number 

In [1]:
number = int(input("Enter the number :"))
count=0
while number>0:
    number//=10
    count+=1
print(count)

Enter the number :4444
4


# Program to find ASCII value   /* Use ord() function
#Note: ASCII of SPACE is 32

In [2]:
wordOrLetter  = input('Enter the words/Letter :')
for char in wordOrLetter:
    print("\nASCII code of {} is {}".format(char,ord(char)))

Enter the words/Letter :Vishwa K Angolkar

ASCII code of V is 86

ASCII code of i is 105

ASCII code of s is 115

ASCII code of h is 104

ASCII code of w is 119

ASCII code of a is 97

ASCII code of   is 32

ASCII code of K is 75

ASCII code of   is 32

ASCII code of A is 65

ASCII code of n is 110

ASCII code of g is 103

ASCII code of o is 111

ASCII code of l is 108

ASCII code of k is 107

ASCII code of a is 97

ASCII code of r is 114


In [13]:
import handcalcs.render #handcalcs is module to render code automatically_in_latin 
from math import sqrt

In [14]:
%%render
a="Vishwa4444"
b=44
c=444
d=4444

<IPython.core.display.Latex object>

#Program to convert number to words #using num2words module

In [17]:
from num2words import num2words
#print(num2words(123))
str = input("Enter the number :")
print("\n\n{} in words is {}".format(str,num2words(str)))

Enter the number :4444


4444 in words is four thousand, four hundred and forty-four


#Program to take in arrays and print 

In [12]:
from array import *
list = array('i',[1,2,3,4,5])
print(list.buffer_info())       #will give the size or length of array
list.append(6)
print(list)
list.reverse()
print(list)
for i in range(len(list)):
    print(list[i],end=" ")
    

(2545189666480, 5)
array('i', [1, 2, 3, 4, 5, 6])
array('i', [6, 5, 4, 3, 2, 1])
6 5 4 3 2 1 

#Program to take array then create a newArray and then take square of the given array of each integer  

In [20]:
from array import *
list = array('i',[12,32,44,56,78])
#for i in range(len(list)):
#    print(list[i],end=" ")
newarr = array(list.typecode,(a*a for a in list))
for i in range(len(newarr)):
    print(newarr[i],end=" ")

144 1024 1936 3136 6084 

#This gives all the builtins in python

In [4]:
print(dir(__builtins__))

['ArithmeticError', 'AssertionError', 'AttributeError', 'BaseException', 'BlockingIOError', 'BrokenPipeError', 'BufferError', 'BytesWarning', 'ChildProcessError', 'ConnectionAbortedError', 'ConnectionError', 'ConnectionRefusedError', 'ConnectionResetError', 'DeprecationWarning', 'EOFError', 'Ellipsis', 'EnvironmentError', 'Exception', 'False', 'FileExistsError', 'FileNotFoundError', 'FloatingPointError', 'FutureWarning', 'GeneratorExit', 'IOError', 'ImportError', 'ImportWarning', 'IndentationError', 'IndexError', 'InterruptedError', 'IsADirectoryError', 'KeyError', 'KeyboardInterrupt', 'LookupError', 'MemoryError', 'ModuleNotFoundError', 'NameError', 'None', 'NotADirectoryError', 'NotImplemented', 'NotImplementedError', 'OSError', 'OverflowError', 'PendingDeprecationWarning', 'PermissionError', 'ProcessLookupError', 'RecursionError', 'ReferenceError', 'ResourceWarning', 'RuntimeError', 'RuntimeWarning', 'StopAsyncIteration', 'StopIteration', 'SyntaxError', 'SyntaxWarning', 'SystemError

#Program to interchange first letter with the last letter in the list

In [6]:
list =[]
size = int(input("Enter the size of array :"))
for i in range(size):
    list.append(int(input()))
print(list)
a=list[0]
b=list[-1]
list[-1]=a
list[0]=b
print(list)

Enter the size of array :4
1
2
3
4
[1, 2, 3, 4]
[4, 2, 3, 1]


##### program to find the second largest number in the list

In [26]:
list=[12,13,13334,185,16,678]
list.sort()    #python uses Tim sort algorithm to sort.
list.pop()
for i in range(len(list)):
    largest= 12
    if list[i]>=largest:
        largest=list[i]
print(largest)
    


678


#program to separate vowels fro the consonants

In [10]:
str = input("Enter the string :")
print("Vowels are :")
for char in str:
    if char in "aeiouAEIOU":
        print(char,end=",")
print("\n\nConsonants are :")
for char in str:
    if char not  in "aeiouAEIOU":
        print(char,end=",")
         
    
        


Enter the string :vishwa
Vowels are :
i,a,

Consonants are :
v,s,h,w,

In [1]:
list=[12,34,2,67,33,32,233,233]
list.sort()
print(list)
list.pop()
print(list)

[2, 12, 32, 33, 34, 67, 233, 233]
[2, 12, 32, 33, 34, 67, 233]


In [2]:
str = input("Ennter a string to reverse :") 
#[start:stop:step]
print(str[::-1])

Ennter a string to reverse :vishwa k angolkar
raklogna k awhsiv


#program to find the maximun number of the handshakes done by a person in a room 

In [1]:
n=int(input("Enter the number of persons :"))
maxHandShakes = int(n*(n-1)/2)
print("max no. of handshakes possible for",n,"people are : ",maxHandShakes)

Enter the number of persons :4
max no. of handshakes possible for 4 people are :  6


#Program to print only the duplicate elements

In [16]:
from collections import Counter
list =[1,1,2,3,4,5,5]
emp_list=[]
d = Counter(list)
print(d)
for item in d:
    print(d[item])
    print(item)


Counter({1: 2, 5: 2, 2: 1, 3: 1, 4: 1})
2
1
1
2
1
3
1
4
2
5


<!-- 
Persistent Coding Questions in 2022 With Solutions:-
Persistent Coding Test Question 1
Sajith loves numbers and coding. His dad gives a task to write a code to find the nth largest number in an array.

Input 

The inputs have : First the count of integers and the second n value
The second line of the input has the list of integers that he needs to do the operations upon.
Output

You have to print the integer representing the nth largest out of the numbers given by his father
 

Constraints

0 < value n <=  count of integers <= 10^6
-10^6 <= each integer <= 10^6
0 <= i < count of integers
 

Example

Input

5  3
11 -1 – 4 12 -6

Output 

-1

Explanation :

-1 is 3rd largest
 -->

# Persistent Coding Questions in 2022 With Solutions:-
#Persistent Coding Test Question 1
Sajith loves numbers and coding. His dad gives a task to write a code to find the nth largest number in an array.

Input 

The inputs have : First the count of integers and the second n value
The second line of the input has the list of integers that he needs to do the operations upon.
Output

You have to print the integer representing the nth largest out of the numbers given by his father
 

Constraints

0 < value n <=  count of integers <= 10^6
-10^6 <= each integer <= 10^6
0 <= i < count of integers
 

Example

Input

5  3
11 -1 – 4 12 -6

Output 

-1

Explanation :

-1 is 3rd largest

In [2]:
size = int(input("Enter the size of array :"))
larNo = int(input("Enter the Largest Number to be Picked from the list:"))
print("Enter ",size," numbers horizontally")
lis = [int(num) for num in input().split(" ",size-1)]
print(lis)
lis.sort()
print("The ",larNo,"largest number is : " ,lis[-larNo])


Enter the size of array :5
Enter the Largest Number to be Picked from the list:3
Enter  5  numbers horizontally
11 -1 -4 12 -6
[11, -1, -4, 12, -6]
The  3 largest number is  : -1


#Persistent Coding Test Question 2
Flipkart has been infected with a virus, the virus works this way: Each user that has been infected with the virus is traced by a special algorithm and a hint that the virus gives. The virus leaves a hint number N.

Flipkart is able to identify the user ID of the virus by this N number as the user ID works as a series : Each number in the series is the sum of the last three numbers in the series. The first three numbers in the series are 0, 0, 1 always.

Write a program to identify the user ID infect based on N value checked from the logs of the system

Input

The input contains the N value left by the virus and found by engineers in the logs of the system
Output

Print the userID of the infect user.
 

Example

Input

11

Output 

81

 

Example

Input

16

Output 

1705

In [2]:
N = int(input("Enter the Number :"))
n1=0
n2=0
n3=1
list=[]
for i in range(N-3):
    n4=n3+n2+n1
    n1,n2,n3=n2,n3,n4
    
    list=[n4]
print(list[0])

Enter the Number :40
3831006429


#List Comprehension

In [5]:
list = [x for x in range(10) if x%2==0]
print(list)



[0, 2, 4, 6, 8]


In [2]:
print("Enter the elements of row :")
list = [[i,j] for i in range(3) for j in range(3) ]
print(list)

Enter the elements of row :
[[0, 0], [0, 1], [0, 2], [1, 0], [1, 1], [1, 2], [2, 0], [2, 1], [2, 2]]


In [5]:
n = int(input())
arr = (map(int, input().split())
print(arr)

SyntaxError: invalid syntax (Temp/ipykernel_32896/1923960478.py, line 3)

#program to find second largest element

In [5]:
list1=[]
list2=[5,4,1,1,2,3,4,5,5]
list2.sort()
for i in list2:
    if i not in list1:
        if i in list2:
            list1.append(i)
print(list1[-2])








4


In [1]:
friends =['Rameez','chetan','rashid','sumer','akash','tushar','Vinayak','Anvita','smita','pranav','rony','bhaskar','rohit']
addOn = [x for x in friends if 'v' in x] 
print(friends)
print(addOn)

['Rameez', 'chetan', 'rashid', 'sumer', 'akash', 'tushar', 'Vinayak', 'Anvita', 'smita', 'pranav', 'rony', 'bhaskar', 'rohit']
['Anvita', 'pranav']


# https://www.hackerrank.com/challenges/list-comprehensions/problem?isFullScreen=true

In [2]:
    x = int(input())
    y = int(input())
    z = int(input())
    n = int(input())
    list=[[i,j,k] for i in range(x+1) for j in range(y+1) for k in range(z+1) if i+j+k!=n]
    print(list)


1
1
2
4
[[0, 0, 0], [0, 0, 1], [0, 0, 2], [0, 1, 0], [0, 1, 1], [0, 1, 2], [1, 0, 0], [1, 0, 1], [1, 0, 2], [1, 1, 0], [1, 1, 1]]


#Remove dupicates from array

In [1]:
# This function returns new size of modified array.
def removeDuplicates(arr, n):

    # Return, if array is empty or contains a single element
    if n == 0 or n == 1:
        return n

    # Start traversing elements
    j = 0;
    for i in range(0, n-1):

        # If current element is not equal to next element then store that current element
        if arr[i] != arr[i+1]:
            arr[j] = arr[i]
            j += 1

    arr[j] = arr[n-1]
    j += 1

    return j

# Driver code
arr = [10, 20, 20, 30, 40, 40, 40, 50, 50]
n = len(arr)

n = removeDuplicates(arr, n)

# Print updated array
for i in range(n):
    print ("%d"%(arr[i]), end = " ")

10 20 30 40 50 

#Program to convert decimal to binary without using function (bin) or anyother function

In [11]:
def DecimalToBinary(num):
    if num >= 1:
        DecimalToBinary(num // 2)
    print(num % 2, end = '')

    
dec_val = 8
DecimalToBinary(dec_val)

01000

#Persistent Coding Question 3
Given an unsorted array of n elements, find if the element k is present in the array or not. Complete the findNumber function in the editor below.

It has 2 parameters:

An array of integers,arr,denoting the elements in the array.
An integer,k,denoting the element to be searched in the array.
The function must return a string “YES” or “NO” denoting if the element is present in the array or not.

Input Format

The first line contains an integers n,denoting the number of elements in the array arr.
Each line i of the n subsequent lines (where 0<=i<=n) contains an integer describing arri.
The next line contains an integer,k,the elements that needs to be searched.
Constraints

1 <= n <= 105
1 <= arr[i]<= 109
Output Format

The function must return a string “YES” or “NO” denoting if the element is present in the array or not.This is printed to stdout by locked stub code in the editor.

Sample Input 0

5
1
2
3
4
5
1

Sample Output 0

YES

Explanation 0

Given the array =[1,2,3,4,5],we want to find the element 1 if it is present or not. We can find the element 1 at index = 0.Therefore we print “YES”.

Sample Input 1

3
2
3
1
5

Sample Output 1

NO

In [1]:
n = int(input())
arr = []
for i in range(n):
    name = int(input())
    arr.append(name)
k = int(input())
if k in arr:
    print('YES')
else:
    print('NO')

5
6
8
8
8
8
7
NO


In [11]:
l1=["vishwa","aman"]
l2=["arav","aman"]
l3=["bryan","harsh"]
l4=l1+l2+l3
print(l4)
l4 = list(set(l4))
print(l4)

TypeError: unhashable type: 'list'

In [2]:
n,k=map(int,input().split())
exmpmatrix=[]
for i in range(n):
    sr=list(map(str,input().split()))
    exmpmatrix.append(sr)
#print(exmpmatrix)
res = sum(exmpmatrix,[])
res = list(set(res))
print(len(res))


2 2
v d h
v j k
5


In [6]:
xx = [['a',1],['b',2],['c',3],['c',4]]
xx = set(tuple(element) for element in xx)
print(xx)

{('a', 1), ('c', 3), ('b', 2), ('c', 4)}


In [10]:
list_of_lists = [[1,2,3], [4,5,6], [7,8,9]]
fl=sum(list_of_lists, []) # [1,2,3,4,5,6,7,8,9]
print(str(fl))

[1, 2, 3, 4, 5, 6, 7, 8, 9]


In [6]:
#2D matrix

In [7]:
d2=[]
r,c=map(int,input().split())
for i in range(r):
    d2.append(list(map(int,input().split())))
print(d2)


3 5
1 2 3 4 5
11 22 33 44 55
111 222 333 444 555
[[1, 2, 3, 4, 5], [11, 22, 33, 44, 55], [111, 222, 333, 444, 555]]


In [7]:
def cal(limit):
    a,b=0,1
    while a<limit:
        yield a
        a,b=b,a+b
limit=int(input("enter:"))
fibonacci = cal(limit)
for i in fibonacci:
    print(i,end=" ")

enter:100
0 1 1 2 3 5 8 13 21 34 55 89 

In [10]:
n= int(input())
def dtob(n):
        if n>=1:
        dtob(n//2)
    print(n%2 ,end="")
dtob(n)


4
0100

In [11]:
#TCS QUESTION

N=int(input())
list=[]
for i in range(N):
    list.append(int(input()))
print("Given list:" ,str(list))
n=len(list)
j=0
for i in range(n):
    if list[i] != 0:
        list[j],list[i] = list[i], list[j]  
        j += 1
for i in range(n):
    print(list[i],end=" ")

    

6
6
0
5
0
1
9
Given list: [6, 0, 5, 0, 1, 9]
6 5 1 9 0 0 

In [18]:
A=[1,2,0,3,0,6]
n=len(list)
j=0
for i in range(6):
    if list[i] != 0:
        list[j], list[i] = list[i], list[j]  # Partitioning the array
        j += 1
print("updated list :",list)

[1, 2, 3, 6, 0, 0]


In [6]:
#explaining friend
def ispa(s):
    return s==s[::-1]
s="wow"
res=ispa(s)
print(res)

True


In [15]:
#BINARY TO DECIMAL
n=input()
res=int(n,2)
print(res)

010101
21


In [29]:
#TCS QUESTION

n=int(input("Enter decimal value :"))

def deciToBin(n):
    return bin(n).replace("0b","")
res= deciToBin(n)
res=list(map(int, str(res)) )   #converted str to list ,Reason:Can toggle list easily.
res=[i^1 for i in res]          #XOR opearation Ex:  0^0=0 , 1^1=0 , 1^0=1 , 0^1=1  

        
   


Enter decimal value :3


TypeError: 'str' object is not callable

#persistent question asked in martian internship test date:17 june 2022

In [6]:
n=int(input())
a = [input() for i in range(n)]

#find frequency of each character
for i in a:  
    d1=dict()
    for c in i:
        if c in d1:
            d1[c]=d1[c]+1 
        else:
            d1[c]=1 
     #sort on basis of items in ascending order       
    d2 = sorted(d1.items())
    d3 = (dict(d2))
    
    max1 = max(d3, key= lambda x: d3[x])
    print(max1)

3
11122334
123244444
552211
[('1', 3), ('2', 2), ('3', 2), ('4', 1)]
1
[('1', 1), ('2', 2), ('3', 1), ('4', 5)]
4
[('1', 2), ('2', 2), ('5', 2)]
1


In [26]:
res=input("Enter binary digits : ")
res=list(map(int, str(res)) )   #converted str to list ,Reason:Can toggle list easily.
res=[i^1 for i in res]          #XOR opearation Ex:  0^0=0 , 1^1=0 , 1^0=1 , 0^1=1  
def result(res):
    for i in res:
        print(i,end=" ")
        return

Enter binary digits : 5


TypeError: 'str' object is not callable

In [9]:
d1 = {'cse':21,'nse':90,'cse':45,'nse':22,'sse':11}
d2 = sorted(d1.items())
print(dict(d2))

{'cse': 45, 'nse': 22, 'sse': 11}


In [5]:
n=int(input())
a = [input() for i in range(n)]
print(a)

3
111123
1213
1232
['111123', '1213', '1232']


#prime numbers

In [7]:
num = int(input("Enter a Prime Number:"))
arr = []

for i in range(2,num):
    flag = 0
    for j in range(2,i):
        if i % j == 0:
            flag = 1 
    if flag == 0:
        arr.append(i)

flag = 0
for i in range(len(arr)):
        for j in range(i+1, len(arr)):
            if arr[i] + arr[j] == num:
                print(str(arr[i])+ " and " +str(arr[j])+ " are the sum of " +str(num))
                flag=1
                break
if flag == 0:
    print("The sum cant be made with prime number:" +str(num))

                
        


Enter a Prime Number:3
The sum cant be made with prime number:3


#Camelcase program , ECOLAB question
 
 #INPUT:
 'hiHelloHowAreYou'
 
 #OUTPUT:
 HI
 hELLO
 hOW
 aRE
 yOU

In [95]:
s = input()
result = s[0].upper() + s[1:]
#print(result)

import re
s=re.findall('[A-Z][a-z]*', result)
listToStr = ' '.join(map(str, s)) 
listToStr = listToStr[0].lower() + listToStr[1:]
print(listToStr)
s1=listToStr.swapcase()
s1=s1.split()
print()
for i in s1:
    print(i)



saveChangesInTheEditor
save Changes In The Editor

SAVE
cHANGES
iN
tHE
eDITOR


In [88]:
for i in range(len(input())):
    print(s[i].upper(),end="")  if s[i].islower() else print("\n",s[i].lower(),end="")

saveChangesInTheEditor
SAVE
 cHANGES
 iN
 tHE
 eDITOR

In [2]:
s = input()
s=s.swapcase()
for i in range(len(s)):
    print(s[i] ,end="")  if s[i].isupper() else print("\n",s[i],end="")

saveChangesInTheEditor
SAVE
 cHANGES
 iN
 tHE
 eDITOR

In [ ]:
b=int(input())
no_J,no_T=list(map(int,input().split()))
list1=[]
list2=[]
for i in range(no_J):
    list1.append(int(input()))
for i in range(no_T):
    list2.append(int(input()))
print(b)
print(list1)
print(list2)

    

In [ ]:
a,b = raw_input().split()

In [1]:
def alternateSort(arr, n):
 
    # Sorting the array
    arr.sort()
 
    # Printing the last element of array
    # first and then first element and then
    # second last element and then second
    # element and so on.
    i = 0
    j = n-1
     
    while (i < j):
     
        print(arr[j], end =" ")
        j-= 1
        print(arr[i], end =" ")
        i+= 1
 
    # If the total element in array is odd
    # then print the last middle element.
    if (n % 2 != 0):
        print(arr[i])
 
 
# Driver code
arr = []
l=int(input())
for i in range(1):
    arr.append((map(int,input().split())))
n=len(arr) 
alternateSort(arr, n)
 
# This code is contributed by
# Smitha Dinesh Semwal

3
1 2 3


In [7]:
lis = [int(num) for num in input().split()]
print(lis)

6 7 9 0
[6, 7, 9, 0]


In [6]:
arr=[]
for i in range(1):
    arr.append(input().split())
print(arr)
arr_1D = sum(arr,[])
print(arr_1D)

SyntaxError: invalid syntax (Temp/ipykernel_908/544444105.py, line 4)

In [9]:
d2=[]
for i in range(1):
    d2.append(list(map(int,input().split())))
print(sum(d2,[]))


1 23 3
[1, 23, 3]


In [10]:
list=[[1,2,3],[4,5,6],[7,8,9]]
print(sum(list,[]))

[1, 2, 3, 4, 5, 6, 7, 8, 9]


In [11]:
n=
numbers= [i*10 for i in range()]
 
print(numbers

SyntaxError: invalid syntax (Temp/ipykernel_908/717181683.py, line 1)

In [1]:
list=[input().split() for i in range(1)]
print(list)

1 2 3 4
[['1', '2', '3', '4']]


In [1]:
p = [int(b) for x in range(n+1) if x >= 1 a,b,c,d = input().split()]
print(p)

SyntaxError: invalid syntax (Temp/ipykernel_21368/1105591025.py, line 1)

In [ ]:
size = int(input("Enter the size of list :"))
print("Enter ",size," numbers horizontally")
lis = [int(num) for num in input().split(" ",size-1)]
print(lis)

In [6]:
N,M,X  = map(int,input().split())
list1 = []
list2 = []
for i in range(N):
        if i == 0:
                list1.append(0)
        else:
                for j in range(N-1):
                        list1.append(list1[j]+M)
                if len(list1)==N:
                        break
print(list1)
for i in range(N):
        if i == 0:
                list2.append(0)
        else:
                for j in range(N-1):
                        list2.append(list2[j]+X)
                if len(list2)==N:
                        break
print(list2)
print(list2[-1]-list1[-1]) # .pop() function by default it points to last index of the list

3 5 5
[0, 5, 10]
[0, 5, 10]
0


In [4]:
N,M,X  = map(int,input().split())
list1 = [i*M for i in range(N)]
print(list1)
list2 = [i*X for i in range(N)]
print(list2)
print(abs(list1[-1]-list2[-1]))   #abs() converts negative number to positive

5 3 10
[0, 3, 6, 9, 12]
[0, 10, 20, 30, 40]
-28


In [4]:
s="hi vishwa here"
s=s.split()
print(s)
s=' '.join(s)
print(s)
print(len(s))

['hi', 'vishwa', 'here']
hi vishwa here
14


#Budget problem persistent

In [7]:
budget = int(input())
n,m=map(int,input().split())
t=list(map(int,input().split()))
j=list(map(int,input().split()))
ans=-1
for i in range(n):
    for k in range(m):
        if t[i]+j[k]<=budget:
            ans=max(ans,t[i]+j[k])
            if ans==budget:
                break
print(ans)

20
3 3
10 5 8 
11 7 4
19


In [ ]:
#form a 2D list
m,n=map(int,input().split())
list=[int(i) for i in input().split(" ",3)]
print(list)

In [ ]:
n,k=map(int,input().split())
e=[]
for i in range(n):
    e.append(list(map(int,input().split(" ",k-1))))
print(e)
for i in range(n):
    for j in range(k):
        if e[i][j]>e[][]
        

In [1]:
unsorted_list=[7,4,3,4,5]
for i in range (len(unsorted_list)):
    for j in range (i,  len(unsorted_list)):
        if unsorted_list[i]> unsorted_list[j]:
            unsorted_list[i], unsorted_list[j] = unsorted_list[j], unsorted_list[i]
print(unsorted_list)

[3, 4, 4, 5, 7]


In [ ]:
list1=[-9,-2,55,43,65,32]
list2=[]
min=-9
while list1:
    for i in list1:
        if i<min:
            min=i
            list2.append(min)
            list1.remove(min)
print(list2)
    
    


In [1]:
#bubble sort Algorithm

In [26]:
def bubblesort(elements):
    swapped = False
    # Looping from size of array from last index[-1] to index [0]
    for n in range(len(elements)-1, 0, -1):
        for i in range(n):
            if elements[i] > elements[i + 1]:
                swapped = True
                # swapping data if the element is less than next element in the array
                elements[i], elements[i + 1] = elements[i + 1], elements[i]       
        if swapped==False:
            # exiting the function if we didn't make a single swap
            # meaning that the array is already sorted.
            return

        
elements = [ 7,12, 18, 18, 39, 72, -85,-1,0]
 
print("Unsorted list is,")
print(elements)
bubblesort(elements)
print("Sorted Array is, ")
print(elements)

Unsorted list is,
[7, 12, 18, 18, 39, 72, -85, -1, 0]
Sorted Array is, 
[-85, -1, 0, 7, 12, 18, 18, 39, 72]


In [3]:
first, *rest = input().split()
print(first)
print(rest)

4 hfgh hgf hjjj  jjj jkkk  k
4
['hfgh', 'hgf', 'hjjj', 'jjj', 'jkkk', 'k']


In [6]:
s="hi hello xns"
s=s.split()
print(len(s))

3


In [10]:
text = "this is the thing"
x = text.replace("is", "")
print(x)

th  the thing


In [11]:
text = "abcdblobefgblobhijk"
sub = "blob"

new_text =''
i=0
last_i = 0
while i <  len(text):
    if text[i:i+len(sub)] == sub:
        new_text += text[last_i:i]
        last_i=i+len(sub)
        i=i+len(sub)
    else:
        i+=1
new_text += text[last_i:i]

print(new_text)

abcdefghijk


In [18]:
input_str = "DivasDwivedi"
result_str = "" 
   
for i in range(len(input_str)): 
    if i != 3: 
        result_str = result_str + input_str[i] 
   
# Printing string after removal   
print ("String after removal of i'th character : " + result_str)

String after removal of i'th character : DivsDwivedi


In [ ]:
str1 = input("Please Enter your Own String : ")
ch = input("Please Enter your Own Character : ")
newch = input("Please Enter the New Character : ")

str2 = ''
for i in range(len(str1)):
    if(str1[i] == ch):
        str2 = str2 + newch
    else:
        str2 = str2 + str1[i]

print("\nOriginal String :  ", str1)
print("Modified String :  ", str2)

#TCS question pattern matching question for a given string

In [39]:
st=input()
st=st.lower()
st1=input()
st1=st1.lower()
print("{} , {}".format(st,st1)) if st.startswith(st1) or st.endswith(st1) else print("{} not contains {}".format(st,st1))


v tcs
tcs
v tcs , tcs


#TCS QUESTION

In [16]:
list=[[10,0],[10,20],[-20,20],[-20,-20],[30,-20],[90,-20],[90,50]]
n=int(input())
ans=list[n-1]
for i in ans:
    print(i,end=" ")

5
30 -20 

#QUESTION SENT BY HUZAIFA ON 28 JUNE 2022

In [15]:
n=int(input())
r=[]
for i in range(n):
    name,digit=input().split()
    digit=int(digit)
    r+=[[name,digit]]    #STORING NAME AND DIGIT IN r[] EXAMPLE:[['CSE', 41], ['SSE', 34]]
    
v1=[]    #making 3 separate 2D list for each CSE , SSE , NSE ....storing in v1,v2,v3
v2=[]
v3=[]
for i in range(len(r)):
    for j in range(len(r[i])):
        if r[i][j]=='CSE':
            v1.append(r[i])
        elif r[i][j]=='SSE':
            v2.append(r[i])
        elif r[i][j]=='NSE':
            v3.append(r[i])

            

v1.sort(key=lambda x:x[1])
v1.reverse()
#print(v1)
v2.sort(key=lambda x:x[1])
v2.reverse()

v3.sort(key=lambda x:x[1])
v3.reverse()

v=v1+v2+v3 #concatinating all 3 list of CSE , SSE , NSE in v
    
ans=sum(v,[])
l=ans[-1]
def f(ans):
    for i in range(len(ans)):
        if i%2==0:
            print(ans[i],end=" ")
        if i%2!=0 and i<len(ans)-1:
            print(ans[i],end=",")
f(ans)            
res=str(f(ans))
print(res)
res1=res.replace("None",str(l))
print((res1))

1
CSE 6


TypeError: 'list' object is not callable

In [6]:
def printCSE(cse, count):
    if count != 0:
        for j in cseStocks:
            print(f"CSE {j}, ", end="")
    else:
        for j in cseStocks:
            if j != cseStocks[-1]:
                print(f"CSE {j}, ", end="")
            else:
                print(f"CSE {j}", end="")


def printSSE(sse, count):
    if count != 0:
        for j in sseStocks:
            print(f"SSE {j}, ", end="")
    else:
        for j in sseStocks:
            if j != sseStocks[-1]:
                print(f"SSE {j}, ", end="")
            else:
                print(f"SSE {j}", end="")


def printNSE(nse):
    for j in nseStocks:
        if j != nseStocks[-1]:
            print(f"NSE {j}, ", end="")
        else:
            print(f"NSE {j}", end="")


n = int(input())
stocks, cseStocks, sseStocks, nseStocks, stockNames = [], [], [], [], []
nseCount, sseCount, cseCount = 0, 0, 0
for i in range(0, n):
    str = input().upper().split()
    stocks.append(str)

for i in range(0, len(stocks)):
    if stocks[i][0] == 'CSE':
        cseStocks.append(int(stocks[i][1]))
        stockNames.append(stocks[i][0])
    elif stocks[i][0] == 'SSE':
        sseStocks.append(int(stocks[i][1]))
        stockNames.append(stocks[i][0])
    elif stocks[i][0] == 'NSE':
        nseStocks.append(int(stocks[i][1]))
        stockNames.append(stocks[i][0])
    else:
        break

cseStocks.sort(reverse=True)
sseStocks.sort(reverse=True)
nseStocks.sort(reverse=True)

for i in stockNames:
    if i == 'CSE':
        cseCount += 1
    elif i == 'SSE':
        sseCount += 1
    elif i == 'NSE':
        nseCount += 1
    else:
        break

if sseCount == 0:
    printCSE(cseStocks, sseCount)
    printNSE(nseStocks)
elif nseStocks == 0:
    printCSE(cseStocks, sseCount)
    printSSE(sseStocks, nseCount)
elif cseStocks == 0:
    printSSE(sseStocks, nseCount)
    printNSE(nseStocks)
else:
    printCSE(cseStocks, sseCount)
    printSSE(sseStocks, nseCount)
    printNSE(nseStocks)


3
CSE 65
NSE 76
SSE 7
CSE 65, SSE 7, NSE 76

In [9]:
n=int(input())
r=[]
for i in range(n):
    name,digit=input().split()
    digit=int(digit)
    r+=[[name,digit]]    #STORING NAME AND DIGIT IN r[] EXAMPLE:[['CSE', 41], ['SSE', 34]]
    
v1=[]    #making 3 separate 2D list for each CSE , SSE , NSE ....storing in v1,v2,v3
v2=[]
v3=[]
for i in range(len(r)):
    for j in range(len(r[i])):
        if r[i][j]=='CSE':
            v1.append(r[i])
        elif r[i][j]=='SSE':
            v2.append(r[i])
        elif r[i][j]=='NSE':
            v3.append(r[i])

            

v1.sort(key=lambda x:x[1])
v1.reverse()
#print(v1)
v2.sort(key=lambda x:x[1])
v2.reverse()

v3.sort(key=lambda x:x[1])
v3.reverse()

v=v1+v2+v3 #concatinating all 3 list of CSE , SSE , NSE in v
    
ans=sum(v,[])
ans=list(map(str,ans))
l=ans[-1]
def f(ans):
    for i in range(len(ans)):
        if i%2==0:
            print(ans[i],end=" ")
        if i%2!=0 and i<len(ans)-1:
            print(ans[i],end=",")
res=str(f(ans))
print(res.replace("None",l))

1
CSE 56


TypeError: 'list' object is not callable

In [17]:
ans=['CSE', 41, 'SSE', 34]
#ans=list(map(str,ans))
l=ans[-1]
def f(ans):
    for i in range(len(ans)):
        if i%2==0:
            print(ans[i],end=" ")
        if i%2!=0 and i<len(ans)-1:
            print(ans[i],end=",")
f(ans)            


CSE 41,SSE 

In [2]:
n=int(input())
r=[]
for i in range(n):
    name,digit=input().split()
    digit=int(digit)
    r+=[[name,digit]]    #STORING NAME AND DIGIT IN r[] EXAMPLE:[['CSE', 41], ['SSE', 34]]
    
v1=[]    #making 3 separate 2D list for each CSE , SSE , NSE ....storing in v1,v2,v3
v2=[]
v3=[]
for i in range(len(r)):
    for j in range(len(r[i])):
        if r[i][j]=='CSE':
            v1.append(r[i])
        elif r[i][j]=='SSE':
            v2.append(r[i])
        elif r[i][j]=='NSE':
            v3.append(r[i])

            

v1.sort(key=lambda x:x[1])
v1.reverse()
#print(v1)
v2.sort(key=lambda x:x[1])
v2.reverse()

v3.sort(key=lambda x:x[1])
v3.reverse()

v=v1+v2+v3 #concatinating all 3 list of CSE , SSE , NSE in v
    
ans=sum(v,[])
ans=list(map(str,ans))
l=ans[-1]
def f(ans):
    for i in range(len(ans)):
        if i%2==0:
            print(ans[i],end=" ")
        if i%2!=0 and i<len(ans)-1:
            print(ans[i],end=",")
res=str(f(ans))
print(res.replace("None",l))

3
CSE 76
NSE 7
CSE 76
CSE 76,CSE 76,NSE 7


In [3]:
n=int(input())
r=[]
for i in range(n):
    name,digit=input().split()
    digit=int(digit)
    r+=[[name,digit]]    #STORING NAME AND DIGIT IN r[] EXAMPLE:[['CSE', 41], ['SSE', 34]]
    
v1=[]    #making 3 separate 2D list for each CSE , SSE , NSE ....storing in v1,v2,v3
v2=[]
v3=[]
for i in range(len(r)):
    for j in range(len(r[i])):
        if r[i][j]=='CSE':
            v1.append(r[i])
        elif r[i][j]=='SSE':
            v2.append(r[i])
        elif r[i][j]=='NSE':
            v3.append(r[i])

            

v1.sort(key=lambda x:x[1])
v1.reverse()
#print(v1)
v2.sort(key=lambda x:x[1])
v2.reverse()

v3.sort(key=lambda x:x[1])
v3.reverse()

v=v1+v2+v3 #concatinating all 3 list of CSE , SSE , NSE in v
    
ans=sum(v,[])
ans=list(map(str,ans))
l=ans[-1]
def f(ans):
    for i in range(len(ans)):
        if i%2==0:
            print(ans[i],end=" ")
        if i%2!=0 and i<len(ans)-1:
            print(ans[i],end=",")
res=str(f(ans))
print(res.replace("None",l))

1
CSE 4
CSE 4


#Print the word which has maximum length in given string

In [25]:
str="The richest monument of India is Taj Mahal "
str1=str.split()
print(str1)
word_at_loc = 0
for i in range(len(str1)):
    if len(str1[i])>word_at_loc:
        word_at_loc=len(str1[i])
print(word_at_loc)
    


['The', 'richest', 'monument', 'of', 'India', 'is', 'Taj', 'Mahal']
8


In [27]:
l1=[['CSE',44],['CSE',43]]
l2=[['SSE',56],['SSE',67]]
l3=[['NSE',87],['NSE',89]]

new_list=l1+l2+l3
print(new_list)

[['CSE', 44], ['CSE', 43], ['SSE', 56], ['SSE', 67], ['NSE', 87], ['NSE', 89]]


In [32]:
r=[['CSE', 44, 3], ['CSE', 43,98], ['CSE', 56,99]]
r.sort(key=lambda x:x[2])
#r.reverse()
print(r)

[['CSE', 44, 3], ['CSE', 43, 98], ['CSE', 56, 99]]


In [1]:
def f(n):
    for x in range(n):
        yield x**3
res =[i for i in f(5)]

In [7]:
def get_bits(length,byte):
    part_1 = "abcdefgh"
    part_2 = 'ijklmnop'
    
    if(byte[0] == '0'):
        part_1_len = len(part_1)
        if (byte[1] == "0"):
            part_1_len -= length
            if (byte[2] == "0"):
                part_1_len -= (length//2)
                if (byte[3] == "1"):
                    print(part_1[1])
                else:
                    print(part_1[0])
                
    # print(length,byte)
    
length_1 = 4
byte_1 = "0001" 

get_bits(length,byte)

SyntaxError: invalid non-printable character U+00A0 (Temp/ipykernel_6000/34929275.py, line 2)

In [1]:
n=input()
n=list(n)
if n[0]=='-':
    j=n.pop(0)
    k=n[::-1]
    k.insert(0,j)
    print(*k,sep="")
else:
    k=n[::-1]
    m=''.join(k)
    f=m.replace('0','')
    print(f)

745333
333547


In [1]:
n=int(input())
if n>0:
    s1=str(n)
    s2=s1[::-1]
    if int(s2)>65535:
        print("w")
        exit()
    print(int(s2))
else:
    n2=abs(n)
    n3=str(n2)
    

7
7


#flip  1 digit and find the max number possible

In [54]:
n=int(input())
n1=str(n)
max=0
for i in n1:
    if max<int(i):
        max=int(i)
n=str(n)
r1=n.replace(n[0],str(max))
print(int(''.join(r1)))


    

    

123
323
